backend/src/ee/services/audit-log/audit-log-queue.ts

```diff
@@ -42,88 +51,144 @@ export const auditLogQueueServiceFactory = ({
   };
 
   queueService.start(QueueName.AuditLog, async (job) => {
-    const { actor, event, ipAddress, projectId, userAgent, userAgentType } = job.data;
-    let { orgId } = job.data;
-    const MS_IN_DAY = 24 * 60 * 60 * 1000;
-    let project;
-
-    if (!orgId) {
-      // it will never be undefined for both org and project id
-      // TODO(akhilmhdh): use caching here in dal to avoid db calls
-      project = await projectDAL.findById(projectId as string);
-      orgId = project.orgId;
+    if (job.name === QueueJobs.AuditLog && typeof job.data !== "undefined") {
+      const { actor, event, ipAddress, projectId, userAgent, userAgentType } = job.data;
+      let { orgId } = job.data;
+      const MS_IN_DAY = 24 * 60 * 60 * 1000;
+      let project;
+
+      if (!orgId) {
+        // it will never be undefined for both org and project id
+        // TODO(akhilmhdh): use caching here in dal to avoid db calls
+        project = await projectDAL.findById(projectId as string);
+        orgId = project.orgId;
+      }
+
+      const plan = await licenseService.getPlan(orgId);
+      if (plan.auditLogsRetentionDays === 0) {
+        // skip inserting if audit log retention is 0 meaning its not supported
+        return;
+      }
+
+      // For project actions, set TTL to project-level audit log retention config
+      // This condition ensures that the plan's audit log retention days cannot be bypassed
+      const ttlInDays =
+        project?.auditLogsRetentionDays && project.auditLogsRetentionDays < plan.auditLogsRetentionDays
+          ? project.auditLogsRetentionDays
+          : plan.auditLogsRetentionDays;
+
+      const ttl = ttlInDays * MS_IN_DAY;
+      const auditLog = {
+        actor: actor.type,
+        actorMetadata: actor.metadata,
+        userAgent,
+        projectId,
+        ipAddress,
+        orgId,
+        eventType: event.type,
+        expiresAt: new Date(Date.now() + ttl),
+        eventMetadata: event.metadata,
+        userAgentType,
+        createdAt: new Date(job.timestamp),
+        updatedAt: new Date(job.timestamp)
+      };
+
+      await keyStore.listAppend(KeyStorePrefixes.AuditLogBatch, [JSON.stringify(auditLog)]);
```

Comment:
Issue is complexity would scale up without any benefits. Consider we have multiple list for each org or project. Now we need to keep a check for all these list so that they don't go out of bounds. 

Next is irrespective of whoever is generating it, it will cause write in db. So it doesn't matter to split it up.